### PR TITLE
Visualizer mediators

### PR DESCRIPTION
**Pull request by hnnesv**
OHM-127

Companion PR https://github.com/jembi/openhim-core-js/pull/750

@rcrichton if you could please review? thanks

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -145,6 +146,8 @@
     <script src="bower_components/ng-file-upload-shim/angular-file-upload-shim.js"></script>
     <script src="bower_components/eonasdan-bootstrap-datetimepicker/build/js/bootstrap-datetimepicker.min.js"></script>
     <script src="bower_components/angular-bootstrap-datetimepicker-directive/angular-bootstrap-datetimepicker-directive.js"></script>
+    <script src="bower_components/angular-bootstrap-colorpicker/js/bootstrap-colorpicker-module.js"></script>
+    <script src="bower_components/angular-fullscreen/src/angular-fullscreen.js"></script>
```
Comment (rcrichton) [inline app/index.html]: This is cool :)

File: app/views/partials/user-settings-tabs.html [hunk 2]
```diff
@@ -80,38 +80,56 @@
       <div class="components_div">
         <label style="float: left">
           Components
-          <i ng-show="uiSettings.showTooltips" tooltip-placement="top" tooltip="Select the components you wish to add to the visualizer" class="glyphicon glyphicon-question-sign tooltip-wide"></i>
+          <i ng-show="uiSettings.showTooltips" tooltip-placement="top" tooltip="Select the components you wish to add to the visualizer. Components are shown 'above the line'." class="glyphicon glyphicon-question-sign tooltip-wide"></i>
```
Comment (rcrichton) [inline app/views/partials/user-settings-tabs.html]: I think we should change these referenced to 'above the line' to 'above the openhim-core' or 'below the openhim-core'. New users won't  understand this I don't think.

File: app/scripts/controllers/visualizer.js [hunk 3]
```diff
@@ -2,18 +2,21 @@
 /* global moment: false */
```
Comment (rcrichton) [inline app/scripts/controllers/visualizer.js]: Do we need any new tests for this file?

**Comment by rcrichton** [comment]
@hnnesv this looks good to me, however, I must admit the d3 stuff gets me a bit lost as I'm not familiar with it. I've added a few comments

Am going to testing the UI some more as well.

File: app/config/visualizer.json [hunk 4]
```diff
@@ -15,6 +16,7 @@
   },
   "time": {
     "updatePeriod": 200,
+    "minDisplayPeriod": 100,
```
Comment (rcrichton) [inline app/config/visualizer.json]: Maybe default to 500ms? 100 is too fast for my browser and probably most users.

**Comment by rcrichton** [comment]
It seems we may need to block audits from the events api to. Core is getting spammed.

**Comment by rcrichton** [comment]
i think we may need to do some usability work around the visualiser settings, however, that is a separate issue. So far, it works nicely.

File: app/index.html (hunk 1 above)
Comment (hnnesv) [inline app/index.html]: :)

File: app/scripts/controllers/visualizer.js (hunk 3 above)
Comment (hnnesv) [inline app/scripts/controllers/visualizer.js]: There weren't any existing tests for this, however I added one in. I had issues running npm test in vagrant though (clean vagrant, fresh node_modules install etc.), must be due to new package versions. I've gone ahead and upgraded all dev-deps. The new jshint is enforcing a new rule though so there are quite a few changes stemming there from.

**Comment by hnnesv** [comment]
@rcrichton if you could please review again? it was quite a nightmare upgrading the package verisons; some versions worked in vagrant and not travis and vice versa - but I got it working now.

**Comment by rcrichton** [comment]
Thanks Hannes. LGTM :+1: The jshint changes are probably for the better. Made it a bit harder to review through :)

**Pull request merged**